client/core/core.go

```diff
@@ -843,7 +846,9 @@ func (dc *dexConnection) reconcileTrades(srvOrderStatuses []*msgjson.OrderStatus
 			}
 		}
 		// If we're updating an order from an active state to executed,
-		// canceled, or revoked, return the remaining quantity.
+		// canceled, or revoked, and there are no active matches, return the
+		// locked funding coins and any refund/redeem reserves.
+		trade.maybeReturnCoins()
```

Comment:
A very longest standing bug in dex.  An order being reconciled from epoch -> revoked/canceled status needs it's coins returned.